elf2table/src/lib.rs

```diff
@@ -149,9 +150,30 @@ pub fn get_locations(elf: &[u8]) -> Result<Locations, anyhow::Error> {
 
         ensure!(cursor.next_dfs()?.is_some(), "empty DWARF?");
 
-        while let Some((_, entry)) = cursor.next_dfs()? {
+        let mut segments = vec![];
+        let mut depth = 0;
+        while let Some((delta_depth, entry)) = cursor.next_dfs()? {
+            depth += delta_depth;
+
             // NOTE .. here start the custom logic
-            if entry.tag() == gimli::constants::DW_TAG_variable {
+            if entry.tag() == gimli::constants::DW_TAG_namespace {
+                let mut attrs = entry.attrs();
+
+                while let Some(attr) = attrs.next()? {
+                    match attr.name() {
+                        gimli::constants::DW_AT_name => {
+                            if let gimli::AttributeValue::DebugStrRef(off) = attr.value() {
+                                let s = dwarf.string(off)?;
+                                for _ in (depth as usize)..segments.len() + 1 {
```

Comment:
this is a bit magic but seems to work. I think it would be more correct to record the depth at which the first path segment appears instead of hardcoding a `1` there